### DISCUSSION TITLE
Graphql-ws Client not reconnecting after failed connection

### DISCUSSION THREAD

**Discussion by Ryiski** [enisdenjo/graphql-ws#357] [Q&A] [answered]
**Screenshot**
![Screenshot 2022-04-25 132356](https://user-images.githubusercontent.com/47484379/165071345-cb474454-5693-46bc-8744-86b2705a1139.png)

**Expected Behaviour**
To try and reconnect after failed connection or closed when server is up again from a crash or reboot

**Actual Behaviour**
not trying to reconnecting once its failed to connect

**Further Information**
im trying to get reconnection to work on my client but it not working for some reason and its on svelte, I've it on nextjs and plain react but its the same issue.

I'd love it if anyone could guide me

Here is the client application

https://drive.google.com/file/d/1qf-UZm5UPCZaCC-nzDIRba7kqPB5HrpK/view?usp=sharing

**Comment by enisdenjo** [comment, marked as answer]
Similar to #355.

All non-close-event connection problems are considered fatal by default. Your client seems to be throwing an `Event`, which is not a [WebSocket CloseEvent](https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent).

If you want the client to retry regardless, you can implement your own [`shouldRetry` option](https://github.com/enisdenjo/graphql-ws/blob/master/docs/interfaces/client.ClientOptions.md#shouldretry) and have it return `true` always (or with pre [v5.8.0](https://github.com/enisdenjo/graphql-ws/releases/tag/v5.8.0) the now [deprecated `isFatalConnectionProblem` option](https://github.com/enisdenjo/graphql-ws/blob/master/docs/interfaces/client.ClientOptions.md#isfatalconnectionproblem) and return `false` always).

**Reply by Ryiski** [reply]
Thank you very much, i have tired isFatalConnectionProblem pre [v5.8.0] and it works, but when i updated to v5.8.0 and tired out shouldRetry and set it to true it doesnt work sadly @enisdenjo

**Reply by enisdenjo** [reply]
Ah, I think I know where the issue lies. Fix incoming!

**Reply by Ryiski** [reply]
Cool 👍, Thank you for the quick repones too, you a super hero 😄 🦸

**Reply by enisdenjo** [reply]
Fix is live with v5.8.1, would be great if you can confirm @Adel-ak? Thanks for reporting the bug, and a bigger thanks for the kind words! 😄

**Reply by Ryiski** [reply]
@enisdenjo its a piece of 🍰 , works amazingly and as intended 👍 , thank you again
